crates/ty_server/src/session.rs

```diff
@@ -495,14 +551,21 @@ impl Session {
         new_version: DocumentVersion,
     ) -> crate::Result<()> {
         let position_encoding = self.position_encoding;
-        self.index_mut()
-            .update_text_document(key, content_changes, new_version, position_encoding)
+        self.index_mut().update_text_document(
+            key,
+            content_changes,
+            new_version,
+            position_encoding,
+        )?;
+        self.bump_revision();
+        Ok(())
     }
 
     /// De-registers a document, specified by its key.
     /// Calling this multiple times for the same document is a logic error.
     pub(crate) fn close_document(&mut self, key: &DocumentKey) -> crate::Result<()> {
         self.index_mut().close_document(key)?;
+        self.bump_revision();
         Ok(())
     }
```

Comment:
Why do we need to bump the version when a file is closed? Workspace diagnostics shouldn't change when a file is closed as they would still be visible in the diagnostics panel in an editor.